service_grpc_namespace/src/lib.rs

```diff
@@ -152,22 +153,88 @@ impl namespace_service_server::NamespaceService for NamespaceService {
             namespace: Some(namespace_to_proto(namespace)),
         }))
     }
+
+    async fn update_namespace_service_protection_limit(
+        &self,
+        request: Request<UpdateNamespaceServiceProtectionLimitRequest>,
+    ) -> Result<Response<UpdateNamespaceServiceProtectionLimitResponse>, Status> {
+        let mut repos = self.catalog.repositories().await;
+
+        let UpdateNamespaceServiceProtectionLimitRequest {
+            name: namespace_name,
+            limit_update,
+        } = request.into_inner();
+
+        debug!(%namespace_name, ?limit_update, "updating namespace service protection limit");
+
+        let namespace = match limit_update {
+            Some(LimitUpdate::MaxTables(n)) => {
+                if n == 0 {
+                    return Err(Status::invalid_argument(
+                        "max table limit for namespace must be greater than 0",
+                    ));
+                }
+                repos
+                .namespaces()
+                .update_table_limit(&namespace_name, n)
+                .await
+                .map_err(|e| {
+                    warn!(error=%e, %namespace_name, table_limit=%n, "failed to update table limit for namespace");
+                    status_from_catalog_namespace_error(e)
+                })
+            }
+            Some(LimitUpdate::MaxColumnsPerTable(n)) => {
+                if n == 0 {
+                    return Err(Status::invalid_argument(
+                        "max columns per table limit for namespace must be greater than 0",
+                    ));
+                }
+                repos
+                .namespaces()
+                .update_column_limit(&namespace_name, n)
+                .await
+                .map_err(|e| {
+                    warn!(error=%e, %namespace_name, per_table_column_limit=%n, "failed to update per table column limit for namespace");
+                    status_from_catalog_namespace_error(e)
+                })
+            }
+            None => Err(Status::invalid_argument(
+                "unsupported service protection limit change requested",
+            )),
+        }?;
+
+        info!(namespace_name, namespace_id = %namespace.id, "updated namespace service protection limits");
```

Comment:
I would recommend logging the new limits too:

```suggestion
        info!(
            namespace_name,
            namespace_id = %namespace.id,
            max_tables = namespace.max_tables,
            max_columns = namespace.max_columns_per_table,
            "updated namespace service protection limits"
        );
```